app/views/products/show.html.haml

```diff
@@ -20,10 +21,11 @@
 %strong Description:
 %br 
 = @product.description
-%br
+%br 
 
-%div.mt-3
+%div.mt-3.mb-5
   = link_to 'Back', products_path, class: 'btn btn-primary'
   - if admin?
     = link_to 'Edit', edit_product_path(@product), class: 'btn btn-secondary'
-    = link_to 'Destroy', product_path(@product), class: 'btn btn-danger', data: {method: :delete, confirm: "Are you sure?"}
\ No newline at end of file
+    = link_to 'Destroy', product_path(@product), {:remote => true, 'data-toggle' =>  "modal", 'data-target' => '#modal-window', class: 'btn btn-danger'}
```

Comment:
you don't need any links for opening modal window, make it with a button https://getbootstrap.com/docs/4.0/components/modal/#live-demo